--- a/memcache_pool.h	2009-02-22 17:01:43.000000000 +0100
+++ b/memcache_pool.h	2009-11-04 14:11:39.000000000 +0100
@@ -45,6 +45,24 @@
 #define MMC_SERIALIZED	0x0001
 #define MMC_COMPRESSED	0x0002
 
+#ifndef PHP_WIN32
+// mac os x has no MSG_NOSIGNAL
+// but >= 10.2 comes with SO_NOSIGPIPE
+// SO_NOSIGPIPE is a setsockopt() option
+// and not a send() parameter as MSG_NOSIGNAL
+// OpenBSD has none of the options so we
+// need to ignore SIGPIPE events
+#ifndef MSG_NOSIGNAL
+#define MSG_NOSIGNAL 0
+#ifdef SO_NOSIGPIPE
+#define USE_SO_NOSIGPIPE
+#endif
+#endif
+
+#include <sys/errno.h>
+
+#endif
+
 /* use second lowest byte to indicate data type */
 #define MMC_TYPE_STRING	0x0000
 #define MMC_TYPE_BOOL	0x0100
